src/modules/rabbitmq/factories/RabbitMQQueueFactory.ts

```diff
@@ -30,6 +33,7 @@ export default class RabbitMQQueueFactory {
         return new InstanceEventQueue(
             this.connectionManager,
             queueName,
+            this.config.get('rabbitmq.topicExchange'),
```

Comment:
Not quite understanding why we've abstracted the exchange into an argument, this caused a bug where this particular line was the wrong exchange type, which wouldn't have been possible previously as we were pulling from the config object.

Could you explain why here we're creating an construction injection yet they're different queue types, and in other places we're not?

Reply:
Before we accessed a property, now we fetch a config value. This seemed more appropriate, as there is more overhead now.

Reply:
Sure, but couldn't we pull that config item out of the config in the class directly rather than being passed in?

Reply:
Why should the `InstanceEventQueue` have a dependency on the `ConfigService`? It just needs a name nothing more. Limiting the spread of dependencies will be better in the long term.